src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
I think he's referring to the last branch, where you're calling `value.to_i64?` and `value.to_f64?` where you're parsing a number just to throw it away and (maybe) parse it again. This is bad. In most cases, this method is only used to check if the plain scalar has a specific value. This could be implemented directly. It's also a simplification: In `Nil.new` you only need to know, if it is a value of `nil`, there is no need to check if might be `true` or a number. `String.to_yaml` is the only place where the current behaviour - a full check over all possible plain values - is needed.